### PR TITLE
lca cnf: Add ibgu tests

### PR DESCRIPTION
**Pull request by josclark42**
This PR updates three IBU tests to use the IBGU introduced in the 4.17 LCA Enhancements epic.

### CODE REVIEW

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/prep-abort-test.go [hunk 1]
```diff
@@ -33,81 +31,39 @@ var _ = Describe(
...
-				prepCguBuilder.Definition.Spec.Enable = ptr.To(true)
+			By("Creating IBGU and monitoring IBU status to report completed", func() {
 
-				prepCguBuilder, err := prepCguBuilder.Create()
-				Expect(err).ToNot(HaveOccurred(), "Failed to create prep CGU.")
+				newIbguBuilder = ibgu.NewIbguBuilder(cnfinittools.TargetHubAPIClient,
+					tsparams.IbguName, tsparams.IbguNamespace).
+					WithClusterLabelSelectors(tsparams.ClusterLabelSelector).
+					WithOadpContent("oadp-cm", "ztp-group").
+					WithSeedImageRef("registry.kni-qe-18.lab.eng.tlv2.redhat.com:5000/ibu/seed:4.17.0-rc.1", "4.17.0-rc.1").
```
Comment (mcornea) [inline tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/prep-abort-test.go:53]: I think we need to load the seed image/version and oadp cm name/namespace as config variables as in https://github.com/openshift-kni/eco-gotests/pull/227/files#diff-6278a8b3de0a7acdaf56dda60e1d478bccfe97bfe808d762c352e5d3716ec039 ?

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go [hunk 2]
```diff
@@ -111,52 +83,19 @@ var _ = Describe(
...
-				upgradeCguBuilder := cgu.NewCguBuilder(TargetHubAPIClient,
-					tsparams.UpgradeCguName, tsparams.IbuCguNamespace, 1).
-					WithCluster(tsparams.TargetSnoClusterName).
-					WithManagedPolicy(tsparams.UpgradePolicyName).
-					WithCanary(tsparams.TargetSnoClusterName)
-				upgradeCguBuilder.Definition.Spec.Enable = ptr.To(true)
+				newIbguBuilder = ibgu.NewIbguBuilder(TargetHubAPIClient,
+					tsparams.IbguName, tsparams.IbguNamespace).
+					WithClusterLabelSelectors(tsparams.ClusterLabelSelector).
+					WithOadpContent("oadp-cm", "ztp-group").
```
Comment (mcornea) [inline tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go:91]: Ditto, we need these loaded as config parameters.

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/internal/tsparams/upgradevars.go [hunk 3]
```diff
@@ -49,4 +49,7 @@ var (
 
 	// TargetSnoClusterName is the name of target sno cluster.
 	TargetSnoClusterName string
+
+	// ClusterLabelSelector is the cluster label passed to IBGUs.
+	ClusterLabelSelector = map[string]string{"CITarget": "true"}
```
[the remark this answers is not in the document]
Comment (josclark42) [reply tests/lca/imagebasedupgrade/cnf/upgrade-talm/internal/tsparams/upgradevars.go:54]: Thanks, @mcornea This was added for testing only when I had two spokes. I will revert it to use common.

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/prep-abort-test.go (hunk 1 above)
Comment (josclark42) [reply tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/prep-abort-test.go]: Updated- thank you.

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go (hunk 2 above)
Comment (josclark42) [reply tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go]: Updated- thank you.

Comment (achuzhoy) [reply tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go]: The NS probably shouldn't be hardcoded to ztp-group.
Look at this thread: https://redhat-internal.slack.com/archives/C05JHD9QYTC/p1726797433687639

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/upgrade-abort-test.go [hunk 4]
```diff
@@ -36,120 +32,63 @@ var _ = Describe(
...
-			})
-
-			By("Creating, and enabling ibu upgrade CGU", func() {
-				upgradeCguBuilder := cgu.NewCguBuilder(cnfinittools.TargetHubAPIClient,
-					tsparams.UpgradeCguName, tsparams.IbuCguNamespace, 1).
-					WithCluster(tsparams.TargetSnoClusterName).
-					WithManagedPolicy(tsparams.UpgradePolicyName).
-					WithCanary(tsparams.TargetSnoClusterName)
-				upgradeCguBuilder.Definition.Spec.Enable = ptr.To(true)
+				newIbguBuilder := ibgu.NewIbguBuilder(cnfinittools.TargetHubAPIClient,
```
Comment (mcornea) [inline tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/upgrade-abort-test.go:71]: Before  creating the abort IBGU maybe we should delete the upgrade IBGU to ensure they're not colliding and avoid  potential race conditions?

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go (hunk 2 above)
Comment (josclark42) [reply tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/rollback-after-failed-upgrade-test.go]: @achuzhoy Thanks- this was changed to use the oadp cm name/namespace variable passed from the CI profile

File: tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/upgrade-abort-test.go (hunk 4 above)
Comment (mcornea) [reply tests/lca/imagebasedupgrade/cnf/upgrade-talm/tests/upgrade-abort-test.go:71]: @josclark42 tnx, if it's expected then looks good.

**Pull request merged**